### PR TITLE
fix: cleanup USE_FB2 macros.

### PR DESCRIPTION
**Pull request by romange**
1. Move uring source files under util/fibers.
2. Now third_party package builds with the same cxx compiler.

### CODE REVIEW

File: util/fibers/submit_entry.h [hunk 1]
```diff
@@ -0,0 +1,206 @@
+// Copyright 2021, Beeri 15.  All rights reserved.
+// Author: Roman Gershman (romange@gmail.com)
+//
```
Comment (romange) [inline util/fibers/submit_entry.h:3]: just moved a file

File: util/fibers/uring_file.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
+// Copyright 2021, Beeri 15.  All rights reserved.
+// Author: Roman Gershman (romange@gmail.com)
+//
+
```
Comment (romange) [inline util/fibers/uring_file.h:4]: just moved a file

File: util/fibers/uring_socket.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2023, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
```
Comment (romange) [inline util/fibers/uring_socket.h:3]: just moved a file

File: util/uring/submit_entry.h [hunk 4]
```diff
@@ -1,214 +1,3 @@
...
-  }
-
-#ifndef USE_FB2
-  friend class Proactor;
-#endif
-};
-
-}  // namespace uring
-}  // namespace util
+#include "util/fibers/submit_entry.h"
```
Comment (royjacobson) [inline util/uring/submit_entry.h:3]: are those headers deprecated now? Can you document it (or put a `#pragma message("deprecated")`)?

Comment (romange) [reply util/uring/submit_entry.h:5]: fixed

**Pull request merged**